scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -45,14 +51,34 @@ private static TaskState getInitialStateForStreams(Map<TaskId, SortedSet<StreamI
      * The state is either taken from the Transport if this task was executed before
      * and is now restarted or is created from scratch if the task hasn't executed
      * successfully before.
+     *
+     * Additionally, the state is trimmed according to the table's TTL value.
      */
-    private Stream<Task> createTasksWithState(Map<TaskId, SortedSet<StreamId>> groupedStreams) {
+    private Stream<Task> createTasksWithState(Map<TaskId, SortedSet<StreamId>> groupedStreams) throws ExecutionException, InterruptedException {
         Map<TaskId, TaskState> states = workerConfiguration.transport.getTaskStates(groupedStreams.keySet());
         TaskState initialState = getInitialStateForStreams(groupedStreams, workerConfiguration.queryTimeWindowSizeMs);
+
+        List<TableName> tableNames = groupedStreams.keySet().stream().map(TaskId::getTable).collect(Collectors.toList());
```

Comment:
Wouldn't set make more sense? Is it possible for the same table to appear twice?

Reply:
Changed to `Set`. Even if the same table appeared twice, here we are using `tableNames` to populate a `Map` (so it would be overwritten and duplicates don't matter).